src/main.py

```diff
@@ -150,7 +168,72 @@ def task3(conn):
         print(row)
 
 def task4(conn):
-    print('Run task 4')
+    print('Running task 4\n')
+    
+    # read and process SQL query -- get number of sessions per author
+    # note that according to the eclass forum, the session count includes ALL sessions,
+    # not just distinct ones
+    # discussion: https://eclass.srv.ualberta.ca/mod/forum/discuss.php?d=1140073#p3001027
+    
+    query = open('4a.sql','r')
+    sql = query.read()
+    query.close()
+
+    df = None
```

Comment:
Not C++. df persists beyond the try block so this predeclaration is unnecessary

Reply:
Yeah, I know python doesn't need predecs, just wanted to make it easier for the TAs to follow. I'll remove it.